### PR TITLE
Add hive strategies to oap session

### PR DESCRIPTION
**Pull request by yueawang**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)
1. Add hive related strategies to OapSessionState as it inherits from HiveSessionStage.
2. Create TestOap (just like TestHive in Spark) which takes the role of oap test context.
3. Modify related unit test to see if OapSession works fine.

## How was this patch tested?
mvn test passes.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/TestOap.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+package org.apache.spark.sql
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.internal.SQLConf
+
+
+object TestOap
+  extends TestOapContext(
+  OapSession.builder.config(
```
Comment (carsonwang) [inline src/main/scala/org/apache/spark/sql/TestOap.scala]: nit: need more indent here or move `extends TestOapContext(` to the upper line?

File: src/main/scala/org/apache/spark/sql/hive/OapSessionState.scala [hunk 2]
```diff
@@ -35,18 +35,26 @@ class OapSessionState(sparkSession: OapSession) extends HiveSessionState(sparkSe
 
   override def planner: SparkPlanner = {
     new SparkPlanner(sparkSession.sparkContext, conf, experimentalMethods.extraStrategies)
-    with OapStrategies
+    with HiveStrategies with OapStrategies
```
Comment (carsonwang) [inline src/main/scala/org/apache/spark/sql/hive/OapSessionState.scala]: nit: Indent

**Comment by yueawang** [comment]
For concern
> 2. The test case is not cover #484 . I mean, no regression test.

Yi has already created hive context related test case, I will merge them during rebase as there is only one Hive context allowed.

**Comment by linhong-intel** [comment]
>Yi has already created hive context related test case, I will merge them during rebase as there is only one Hive context allowed.

and .gitignore? then we can merge it.

**Comment by yueawang** [comment]
Done!

**Pull request merged**